src/components/map/PinModal.tsx

```diff
@@ -0,0 +1,71 @@
+import React from "react";
```

Comment:
nit: could you change all these double indents to just a single indent (2 spaces)?

Reply:
> I tested this branch locally, it seems to be really slow for me and it comes back with several errors. Do you get this on your end?

@joanna-chen this change actually depends on some backend work to clean up our get/record_details endpoint (gets a single record upon specifying a record source_id). that's why I've made this PR a draft for now.